content/blog/switching-the-application-stack-from-pern-to-mern/index.md

```diff
@@ -0,0 +1,392 @@
+---
+title: "Switching the application stack from PERN to MERN"
+date: 2020-09-18
+meta_desc: Demonstrating the simplicity, modularity, and reusability of running an application on Kubernetes using Pulumi.
+meta_image: meta.png
+authors: ["vova-ivanov"]
+tags: ["aws", "typescript", "docker", "kubernetes"]
+---
+
+In this blog post we will return to the PERN application we previously [migrated to Kubernetes]({{< relref "/blog/deploying-a-pern-stack-application-to-aws" >}}), and switch out the PostgreSQL database for MongoDB. Although at first glance it might seem like a difficult task, the straigthforward design of Pulumi and Kubernetes allows us to easily transition the application form a PERN stack to a MERN one.
+
+<!--more-->
+
+The main difference between PostgreSQL and MongoDB is the way they store their data: PostgreSQL is a relational database which allows users to create complex tables and relations between values, while MongoDB is a key-value database that stores data in simple collections without any special relations. Because the application takes the form of three distinct components--the client, server, and database--almost all of the changes will occur in the database component. We can in fact reuse most of the existing code, and will be able to change the database without additional hassle.
```

Comment:
```suggestion
The main difference between PostgreSQL and MongoDB is data is stored. PostgreSQL is a relational database that allows users to create complex tables and relations between values. In contrast, MongoDB is a key-value database that stores data in simple collections without any special relations. Because the application takes the form of three distinct components--the client, server, and database--almost all of the changes will occur in the database component. In fact, we can reuse most of the existing code and will be able to change the database without additional hassle.
```